datafusion/src/catalog/schema.rs

```diff
@@ -101,4 +115,70 @@ impl SchemaProvider for MemorySchemaProvider {
         let mut tables = self.tables.write().unwrap();
         Ok(tables.remove(name))
     }
+
+    fn table_exist(&self, name: &str) -> bool {
+        let tables = self.tables.read().unwrap();
+        tables.contains_key(name)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::arrow::datatypes::SchemaRef;
+    use crate::catalog::schema::{MemorySchemaProvider, SchemaProvider};
+    use crate::datasource::TableProvider;
+    use crate::logical_plan::Expr;
+    use crate::physical_plan::ExecutionPlan;
+    use std::any::Any;
+    use std::sync::Arc;
+
+    use async_trait::async_trait;
+
+    #[tokio::test]
+    async fn test_mem_provider() {
+        struct TestTableProvider();
```

Comment:
Is it necessary to create a test provider?
`EmptyTable` seems enough for the test. You could create a dummy table using `EmptyTable::new(Arc::new(Schema::empty()))`.

Reply:
i just refer to [this](https://github.com/apache/arrow-datafusion/blob/7824a8d74093374da8a4f040d23a81b8436b7380/datafusion/src/execution/context.rs#L3469).
Thanks for your comments, the empty table is better.